app/javascript/map.js

```diff
@@ -179,11 +179,21 @@ class Map {
     return marker;
   }
 
-  hydrateLatLng(latElementId, lngElementId) {
+  hydrateLatLng(latElementId, lngElementId, searchBoxId) {
+    let searchBox = new google.maps.places.SearchBox(document.getElementById(searchBoxId));
+
     latId = latElementId;
     lngId = lngElementId;
-    googleMap.addListener('click', e => {
-      this.addDraggableMarker(e.latLng);
+
+    searchBox.addListener('places_changed', () => {
+      let places = searchBox.getPlaces();
+
+      if(places.length == 0) return;
+
+      let location = places[0].geometry.location;
```

Comment:
I think we need to be careful of the use of a variable named `location` because that is a JS keyword and breaks IE... instead use something like `placeLocation` just to disambiguate.